woocommerce/wc-admin
# Add mock product report, helper functions

This PR starts off a "Product Report" page by adding mock data for the table. Eventually this will be pulled from a combo of 2 API endpoints, as the product revenue data endpoint won't contain product name/category/stock etc. See p7bje6-14Q-p2, "The Table"

For now, I'm pretending we've fetched all that & merged it into one array. This is then used to display a table on Analytics > Products.

This PR also sets up a few helper functions for other reports:

- The `onQueryChange` handler is moved to `nav-utils`, so table sorting & pagination can be handled in one place
- Introduces a helper function `getIdsFromQuery` to parse numeric IDs from a comma-separated string in the query (ex: `products=2,6,18` will give us the array `[ 2, 6, 18 ]`)
 
**To test**

- Go to Analytics > Products, make sure the table shows up
- Try "sorting" a table column, the URL should update with the new `order_by` (the table _won't update_ since it's not hooked up to an API)
- Go to Analytics > Revenue, make sure no functionality has changed

Note: this PR is a precursor to adding the "Compare" functionality to the table #94
